sui_core/src/gateway_state.rs

```diff
@@ -53,6 +56,123 @@ const MAX_TX_RANGE_SIZE: u64 = 4096;
+                "Total number of transactions which errored out"
+            )
+            .unwrap(),
+            // total_effects == total transactions finished
+            num_tx_publish: register_int_counter!(
+                "num_tx_publish",
+                "Number of publish transactions",
+            )
+            .unwrap(),
+            num_tx_movecall: register_int_counter!(
+                "num_tx_movecall",
+                "Number of MOVE call transactions",
+            )
+            .unwrap(),
+            num_tx_splitcoin: register_int_counter!(
+                "num_tx_splitcoin",
+                "Number of split coin transactions",
+            )
+            .unwrap(),
+            num_tx_mergecoin: register_int_counter!(
+                "num_tx_mergecoin",
+                "Number of merge coin transactions",
+            )
+            .unwrap(),
+            total_tx_certificates: register_int_counter!(
+                "total_tx_certificates",
+                "Total number of certificates made from validators",
+            )
+            .unwrap(),
+            total_tx_retries: register_int_counter!(
+                "total_tx_retries",
+                "Total number of retries for transactions",
+            )
+            .unwrap(),
+            shared_obj_tx: register_int_counter!(
+                "gateway_shared_obj_tx",
+                "Number of transactions involving shared objects"
+            )
+            .unwrap(),
+            // It's really important to use the right histogram buckets for accurate histogram collection.
+            // Otherwise values get clipped
+            num_signatures: register_histogram!(
+                "num_signatures_per_tx",
+                "Number of signatures collected per transaction",
+                POSITIVE_INT_BUCKETS.to_vec()
+            )
+            .unwrap(),
+            num_good_stake: register_histogram!(
+                "num_good_stake_per_tx",
+                "Amount of good stake collected per transaction",
+                POSITIVE_INT_BUCKETS.to_vec()
+            )
+            .unwrap(),
+            num_bad_stake: register_histogram!(
+                "num_bad_stake_per_tx",
+                "Amount of bad stake collected per transaction",
+                POSITIVE_INT_BUCKETS.to_vec()
+            )
+            .unwrap(),
+            transaction_latency: register_histogram!(
+                "transaction_latency",
+                "Latency of execute_transaction_impl"
+            )
+            .unwrap(),
+        }
+    }
+}
+
+impl Default for GatewayMetrics {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+// One cannot register a metric multiple times.  We protect initialization with lazy_static
+// for cases such as local tests or "sui start" which starts multiple authorities in one process.
+lazy_static! {
+    static ref METRICS: Arc<GatewayMetrics> = Arc::new(GatewayMetrics::new());
+}
```

Comment:
Again we should use OnceCell and not lazy static

Reply:
Ok, would you mind enlightening us as to why using lazy_static is bad?  Thanks